File: S2-Regression/Pt6-RandomForestRegression/random_forest_regression.py
## Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


## Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
x = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values


## Training the Random Forest Regression model on the whole data set
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=10, random_state=0)
regressor.fit(x, y)


## Predicting a new result
y_pred = regressor.predict([[6.5]])


## Visualizing the Random Forest Regression results
x_grid = np.arange(min(x), max(x), 0.01)
x_grid = x_grid.reshape(len(x_grid), 1)
plt.scatter(x, y, color='red')
plt.plot(x_grid, regressor.predict(x_grid), color='blue')
plt.title('Truth or Bluff (Random Forest Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

File: S2-Regression/Pt6-RandomForestRegression/random_forest_regression.R
## Importing the data set
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

## Fitting the Random Forest Regression model into the data set
# install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 500)

## Predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5))

## Visualizing the Random Forest Regression results
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(
    x = dataset$Level,
    y = dataset$Salary
  ), colour = 'red') +
  geom_line(aes(
    x = x_grid,
    y = predict(regressor, data.frame(Level = x_grid))
  ), colour = 'blue') +
  ggtitle('Truth or Bluff (Random Forest Regressor)') +
  xlab('Position Level') +
  ylab('Salary')
